test/test_integration.py

```diff
@@ -54,58 +56,129 @@ def tearDown(self):
+        self.assertEqual(account_id, payment_method[0].account_id)
+        self.assertEqual('__EXTERNAL_PAYMENT__', payment_method[0].plugin_name)
 
         # Tag account as AUTO_INVOICING_OFF
-        # TODO NOT WORKING! [Bad Request: POST http://localhost:8080/1.0/kb/accounts/b506e9bc-0afd-495c-a047-a1eada0599be/tags?tagDef=0&tagDef=0&tagDef=0&tagDef=0&tagDef=0&tagDef=0&tagDef=0&tagDef=0&tagDef=-&tagDef=0&tagDef=0&tagDef=0&tagDef=0&tagDef=-&tagDef=0&tagDef=0&tagDef=0&tagDef=0&tagDef=-&tagDef=0&tagDef=0&tagDef=0&tagDef=0&tagDef=-&tagDef=0&tagDef=0&tagDef=0&tagDef=0&tagDef=0&tagDef=0&tagDef=0&tagDef=0&tagDef=0&tagDef=0&tagDef=0&tagDef=2]
-        account_id = 'b506e9bc-0afd-495c-a047-a1eada0599be'
-        account.create_account_tags(account_id, 'test', 'bob', 'lazar')
+        account = killbill.api.AccountApi()
+        tag = ["00000000-0000-0000-0000-000000000002"]
+        account.create_account_tags(account_id, tag, 'test', random_api_key, random_api_secret)
+
+        # Get account tags
+        tags = account.get_account_tags(account_id, random_api_key, random_api_secret)
+        self.assertIsNotNone(tags[0].tag_id)
+        self.assertEqual("00000000-0000-0000-0000-000000000002", tags[0].tag_definition_id)
+        self.assertEqual("AUTO_INVOICING_OFF", tags[0].tag_definition_name)
 
         # Create a subscription against plan
-        # TODO NOT WORKING! [ValueError: Invalid value for `product_name`, must not be `None`]
         subscription = killbill.api.SubscriptionApi()
-        body = Subscription(account_id='4eba7df9-f023-4425-bb46-b1a2e6e84fea',
-                            plan_name='sports-monthly',
-                            product_name='Basic',
-                            billing_period='MONTHLY',
-                            price_list='DEFAULT')
-        subscription.create_subscription(body, 'test', 'bob', 'lazar')
+        body = Subscription(account_id=account_id, plan_name='standard-monthly')
+        subscription.create_subscription(body, 'test', random_api_key, random_api_secret)
+
+        # Get account bundles
+        account = killbill.api.AccountApi()
+        bundles = account.get_account_bundles(account_id, random_api_key, random_api_secret)
+        subscription_id = bundles[0].subscriptions[0].subscription_id
+
+        # Get subscription
+        subscription = killbill.api.SubscriptionApi()
+        subscription = subscription.get_subscription(subscription_id, random_api_key, random_api_secret)
+        self.assertEqual('standard-monthly', subscription.plan_name)
+
+        # Get account invoices
+        account = killbill.api.AccountApi()
+        invoices = account.get_invoices_for_account(account_id, random_api_key, random_api_secret)
+        self.assertEqual([], invoices)
+
+        # Remove AUTO_INVOICING_OFF tag
+        account = killbill.api.AccountApi()
+        account.delete_account_tags(account_id, 'test', random_api_key, random_api_secret, tag_def=tag)
+
```

Comment:
In order to make it simple, we could add a sleep `500` mSec -- to make sure Kill Bill core had time to process the invoice. A more sophisticated way would be to rely on the [getQueueEntries](https://github.com/killbill/killbill/blob/work-for-release-0.19.x-doc/jaxrs/src/main/java/org/killbill/billing/jaxrs/resources/AdminResource.java#L158) api.